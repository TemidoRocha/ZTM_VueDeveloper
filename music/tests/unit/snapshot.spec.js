import { shallowMount, RouterLinkStub } from '@vue/test-utils';
import SongItem from '@/components/SongItem.vue';

describe('Snapshot SongItem.vue', () => {
  test('renders correctly', () => {
    const song = {
      docId: 'abs',
      modified_name: 'Olecas',
      display_name: 'Olecas',
      comment_count: 5,
    };

    const wrapper = shallowMount(SongItem, {
      propsData: { song },
      global: { components: { 'router-link': RouterLinkStub } }
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
