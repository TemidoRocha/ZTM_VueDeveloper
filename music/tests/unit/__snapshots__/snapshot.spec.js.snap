// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot SongItem.vue renders correctly 1`] = `
<li
  class="flex justify-between items-center p-3 pl-6 cursor-pointer transition duration-300 hover:bg-gray-50 song-class-id-abs"
  id="song-id-abs"
>
  <div>
    <router-link-stub
      class="font-bold block text-gray-600"
      custom="false"
      to="[object Object]"
    />
    <span
      class="text-gray-500 text-sm"
    >
      Olecas
    </span>
  </div>
  <div
    class="text-gray-600 text-lg"
  >
    <router-link-stub
      custom="true"
      to="[object Object]"
    />
  </div>
</li>
`;
